src/shared/modules/discovery/discoveryDuck.ts

```diff
@@ -171,23 +196,76 @@ export const discoveryOnStartupEpic = (some$: any, store: any) => {
           discovered: onlyTruthy
         })
       }
+      const discoveryEndpoint = getDiscoveryEndpoint(
+        getHostedUrl(store.getState())
+      )
       return Rx.Observable.fromPromise(
         remote
-          .getJSON(getDiscoveryEndpoint(getHostedUrl(store.getState())))
+          .getJSON(action.discoveryURL || discoveryEndpoint)
           // Uncomment below and comment out above when doing manual tests in dev mode to
           // fake discovery response
           //Promise.resolve({
           // bolt: 'bolt://localhost:7687',
-          //neo4j_version: '4.0.3'
+          // neo4j_version: '4.0.3'
           //})
-          .then(result => {
+          .then(async result => {
+            const ssoProviders =
+              result.sso_providers || result.ssoproviders || result.ssoProviders
+            let creds: { username?: string; password?: string } = {}
+            let ssoError: string | undefined
+
+            if (ssoProviders) {
+              authLog('SSO providers found on endpoint')
+              checkAndMergeSSOProviders(ssoProviders, true)
```

Comment:
What is `isLocalHostOrigin` and why are we always passing `true` ?

Reply:
switched to a better name